feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/data/network/blockchain/struct/extrinsic/TransferExtrinsicFactory.kt

```diff
@@ -0,0 +1,187 @@
+package jp.co.soramitsu.feature_wallet_impl.data.network.blockchain.struct.extrinsic
+
+import jp.co.soramitsu.common.utils.SuspendableProperty
+import jp.co.soramitsu.common.utils.hash
+import jp.co.soramitsu.fearless_utils.encrypt.EncryptionType
+import jp.co.soramitsu.fearless_utils.encrypt.Signer
+import jp.co.soramitsu.fearless_utils.encrypt.model.Keypair
+import jp.co.soramitsu.fearless_utils.scale.invoke
+import jp.co.soramitsu.feature_account_api.domain.model.Node
+import jp.co.soramitsu.feature_wallet_impl.data.network.blockchain.extrinsics.signExtrinsic
+import jp.co.soramitsu.feature_wallet_impl.data.network.blockchain.response.RuntimeVersion
+import jp.co.soramitsu.feature_wallet_impl.data.network.blockchain.struct.Signature
+import java.math.BigInteger
+
+class TransferExtrinsic(
+    val senderId: ByteArray,
+    val recipientId: ByteArray,
+    val amountInPlanks: BigInteger,
+    val index: Pair<UByte, UByte>,
+    val hash: String
+)
+
+class EncodeExtrinsicParams(
+    val senderId: ByteArray,
+    val recipientId: ByteArray,
+    val amountInPlanks: BigInteger,
+    val nonce: BigInteger,
+    val runtimeVersion: RuntimeVersion,
+    val networkType: Node.NetworkType,
+    val encryptionType: EncryptionType,
+    val genesis: ByteArray
+)
+
+class TransferExtrinsicFactory(
+    private val isUpgradedToDualRefCount: SuspendableProperty<Boolean>,
```

Comment:
Oh, don't forget not to call in main thread ;)

Reply:
Sure)